### PR TITLE
Extract schannel connect step1

### PR DESCRIPTION
**Pull request by ebejan**
Extracting code out of Schannel_connect_step1 to
1. reduce complexity
2. prepare for #6718

### CODE REVIEW

**Comment by ebejan** [comment]
@mback2k would you take a look at this one?

**Review by mback2k — APPROVED** [review]

**Comment by mback2k** [comment]
Merging once AppVeyor tests have finished and passed.

**Comment by ebejan** [comment]
I am sure that i have fixed the failing appveyor failure. This could be using cached old version.

File: lib/vtls/schannel.c [hunk 1]
```diff
@@ -515,326 +848,9 @@ schannel_connect_step1(struct Curl_easy *data, struct connectdata *conn,
...
-      case SEC_E_NO_CREDENTIALS:
-      case SEC_E_SECPKG_NOT_FOUND:
-      case SEC_E_NOT_OWNER:
-      case SEC_E_UNKNOWN_CREDENTIALS:
-      case SEC_E_INTERNAL_ERROR:
-      default:
-        return CURLE_SSL_CONNECT_ERROR;
-      }
+    result = schannel_acquire_credential_handle(data, conn, sockindex);
+    if(result != CURLE_OK) {
```
Comment (ebejan) [inline lib/vtls/schannel.c:852]: This part is already changed. AppVeyor is failing somehow here.

Comment (mback2k) [reply lib/vtls/schannel.c:852]: AppVeyor is flaky and not failing due to your changes, I am retriggering it until it finishes cleanly. Don't worry.

**Review by mback2k — APPROVED** [review]

**Comment by bagder** [comment]
Can you please set your name as a committer, not a URL?

**Comment by ebejan** [comment]
> Can you please set your name as author, not a URL?

@bagder I am not sure how i do that. I thought i have not put my name anywhere.:(

**Comment by bagder** [comment]
The name is in the commit itself:
~~~
git log
...
Author: https://gitlab.tableausoftware.com <ejanchivdorj@tableau.com>
Date:   Tue Jun 1 13:37:08 2021 -0700

    schannel: move code out of SChannel_connect_step1
~~~

You have a URL set where your name is supposed to be. To set it to something more sensible:
~~~
git config --global user.name "FIRST_NAME LAST_NAME"

**Comment by ebejan** [comment]
> The name is in the commit itself:
> 
> ```
> git log
> ...
> Author: https://gitlab.tableausoftware.com <ejanchivdorj@tableau.com>
> Date:   Tue Jun 1 13:37:08 2021 -0700
> 
>     schannel: move code out of SChannel_connect_step1
> ```
> 
> You have a URL set where your name is supposed to be. To set it to something more sensible:
> 
> ```
> git config --global user.name "FIRST_NAME LAST_NAME"
> ```

Done.

**Comment by bagder** [comment]
You then need to amend your commit and make sure git stores your name in there, then force-push it here. Or if you write your name here I can fit it locally in my end before I push.

If you check the commit patch like on this URL: https://patch-diff.githubusercontent.com/raw/curl/curl/pull/7168.patch you see that the `From:` line at the top has a URL instead of a name.

**Comment by ebejan** [comment]
I think i did yesterday followed http://treeindev.net/article/git-change-commit-name, but not too familiar with git. My name is Ebe Janchivdorj if it didn't work. 

> git log shows me:
commit e76a4cd6cb42d148197771a0214bcfb39282e139 (HEAD -> pr/7168, origin/pr/7168)
Author: Ebe Janchivdorj <ejanchivdorj@tableau.com>
Date:   Tue Jun 1 14:01:09 2021 -0700

    Fix white space problem
    
    Build showed that
    
    ./vtls/schannel.c:852:7: warning: if with space (SPACEBEFOREPAREN)
         if (result != CURLE_OK) {
           ^

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**